client/src/controllers/BulkController.test.js

```diff
@@ -0,0 +1,58 @@
+import { Application } from '@hotwired/stimulus';
+import { BulkController } from './BulkController';
+
+describe('BulkController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+    <div data-controller="w-bulk">
+      <input type="checkbox" data-w-bulk-target="all" data-action="w-bulk#toggleAll">
+      <div>
+        <input type="checkbox" data-w-bulk-target="item" disabled data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+      </div>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+    </div>
+    `;
+    const application = Application.start();
+    application.register('w-bulk', BulkController);
+  });
+
+  it('executes the correct action when the select all checkbox is clicked', () => {
+    const allCheckbox = document.querySelector('[data-w-bulk-target="all"]');
+    const toggleAllActionSpy = jest.spyOn(
+      BulkController.prototype,
+      'toggleAll',
+    );
+
+    allCheckbox.click();
+
+    expect(toggleAllActionSpy).toHaveBeenCalled();
+  });
```

Comment:
```suggestion
  it('selects all checkboxes when the select all checkbox is clicked', () => {
    const allCheckbox = document.getElementById('select-all');

    allCheckbox.click();

    expect(
      document
        .getElementById('checkboxes')
        .querySelectorAll(':checked:not(:disabled)').length,
    ).toEqual(2);

    allCheckbox.click();

    expect(
      document
        .getElementById('checkboxes')
        .querySelectorAll(':checked:not(:disabled)').length,
    ).toEqual(0);
  });
```

Avoid writing tests that 'dig into' the implementation, your goal is to test that when a select all is clicked the other checkboxes get selected. The unit test should not care 'how', just that one click leads to other things being checked.